fem/bilinearform_ext.cpp

```diff
@@ -102,23 +102,33 @@ void PABilinearFormExtension::Mult(const Vector &x, Vector &y) const
    Array<BilinearFormIntegrator*> &integrators = *a->GetDBFI();
 
    const int iSz = integrators.Size();
-   if (elem_restrict_lex)
+   if (Device::Allows(Backend::CEED_MASK))
    {
-      elem_restrict_lex->Mult(x, localX);
-      localY = 0.0;
       for (int i = 0; i < iSz; ++i)
       {
-         integrators[i]->AddMultPA(localX, localY);
+         integrators[i]->AddMultPA(x, y);
       }
```

Comment:
Does this work in parallel? What about with AMR?

Reply:
I believe that this work in parallel, `local` here means that it is an E-vector, but I let libCEED do the scatter/gather, that's why I'm giving the L-vectors `x` and `y`.
AMR doesn't work since it is not yet supported by libCEED, I imagine that if all we need is a weight vector for AMR, then it would be straightforward to add AMR to libCEED.

Reply:
I think this works fine even with AMR and in parallel since both `x` and `y` are `GridFunction`-size vectors and the transition to and from E-vector is still a boolean matrix even in the case of AMR. This means that the scatter and gather in libCEED are sufficient. The real parallel and/or AMR interpolation is in the prolongation and restriction operators applied on top of this `Mult` action. These prolongation/restriction operators are defined in the base class `BilinearFormExtension` by the methods `GetProlongation` and `GetRestriction` which just use the respective methods from the internally stored `BilinearForm` pointer `a`.

Reply:
I tried it, AMR works.